kalite/inline/static/js/inline/views.js

```diff
@@ -66,18 +99,25 @@ var ButtonView = BaseView.extend({
                 key = key[0];
                 var value = attribute[key];
                 var newkey = null;
-                var newvalue = null;
+                // var newvalue = null;
 
                 if (key === "text") {
                     newkey = "intro";
-                    newvalue = value;
                 }
                 else if (key === "position") {
                     newkey = "position";
-                    newvalue = value;
+                }
+                else if (key === "step") {
+                    newkey = "step";
                 }
                 else if (key === "before-showing") {
-                    //TO DO: user actions to taken before showing modal
+                    //add the before-showing attribute to a separate
+                    //"before-showing" array to be used in intro.onchange()
+                    console.log("inside before-showing building");
+                    var before_showing_obj = {};
+                    before_showing_obj["element"] = selectorKey;
```

Comment:
Again, this and the logic that depends on it don't conform to [the spec](https://docs.google.com/document/d/11wY1EViZrCKXZkus-7Pf4RMCbEYuaTNSCRa5YCglMMU/edit).